applications/CoSimulationApplication/python_scripts/development_helpers/external_field_solver.py

```diff
@@ -0,0 +1,184 @@
+
+        for i in range(num_coupling_interfaces):
+            coupling_interface_settings = self.settings["coupling_interfaces"][i]
+            sub_model_part_name = coupling_interface_settings["sub_model_part_name"].GetString()
+            comm_name = coupling_interface_settings["comm_name"].GetString()
+
+            receive_interface = False
+            operation = "Exporting"
+            if coupling_interface_settings.Has("receive_interface"):
+                receive_interface = coupling_interface_settings["receive_interface"].GetBool()
+                operation = "Importing"
+
+            self.__CustomPrint(2, colors.cyan(operation)+' coupling-interface "{}" on ModelPart "{}"'.format(comm_name, sub_model_part_name))
+
+            if not self.dry_run:
+                if receive_interface:
+                    # use an aux-modelpart, which will not be used again
+                    KratosCoSim.EMPIRE_API.EMPIRE_API_recvMesh(self.model.CreateModelPart(sub_model_part_name), comm_name)
+                    # TODO maybe restructure and do all receives after all sends
+                else:
+                    KratosCoSim.EMPIRE_API.EMPIRE_API_sendMesh(self.model["ExtSolver."+sub_model_part_name], comm_name)
+            else:
+                self.__CustomPrint(2, colors.magenta('... skipped'))
+
+            self.__CustomPrint(1, colors.cyan("Finished "+operation) + " of CouplingInterfaces")
+
+        # time loop
+        self.__CustomPrint(1, "Starting Solution Loop")
+        for i in range(self.settings["num_steps"].GetInt()):
+            print() # newline
+            self.__CustomPrint(1, colors.bold('Step: {}/{}'.format(i+1, self.settings["num_steps"].GetInt())))
+            self.SolveSolutionStep()
+        self.__CustomPrint(1, "Finished")
+
+    def SolveSolutionStep(self):
+        num_coupling_interfaces = self.settings["coupling_interfaces"].size()
+
+        self.__CustomPrint(1, colors.green("Starting import") + " of CouplingInterfaceData ...")
+
+        for i in range(num_coupling_interfaces):
+            coupling_interface_settings = self.settings["coupling_interfaces"][i]
+            if not coupling_interface_settings.Has("data_field_recv"):
+                continue
+            sub_model_part_name = coupling_interface_settings["sub_model_part_name"].GetString()
+            data_field_settings = coupling_interface_settings["data_field_recv"]
+
+            data_field_name = data_field_settings["data_field_name"].GetString()
+
+            self.__CustomPrint(2, colors.green('Importing')+' data-field "{}" on ModelPart "{}"'.format(data_field_name, sub_model_part_name))
+
+            variables = GenerateVariableListFromInput(data_field_settings["variables"])
+
+            if not self.dry_run:
+                KratosCoSim.EMPIRE_API.EMPIRE_API_recvDataField(self.model["ExtSolver."+sub_model_part_name], data_field_name, *variables)
```

Comment:
a comment about the * here, about what it does, is helpful when reading !

Reply:
done